### PR TITLE
Add prow configuration for ironic-agent-image

### PR DESCRIPTION
**Pull request by dtantsur**
This repository provides means to build an agent container for Ironic.

The CI configuration is taken from ironic-image more or less verbatim.

### CODE REVIEW

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    tag: rhel-8-base-openshift-4.8
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.15
+images:
+- dockerfile_path: Dockerfile.ocp
+  from: ocp_builder_rhel-8-base-openshift-4.8
+  inputs:
```
Comment (elfosardo) [inline ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main.yaml:18]: you don't need inputs in this case, no multi stage image

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main__prevalidation.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    tag: rhel-8-base-openshift-4.8
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.15
+images:
+- dockerfile_path: Dockerfile.ocp
+  from: base
+  inputs:
```
Comment (elfosardo) [inline ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main__prevalidation.yaml:18]: same here, you can remove the inputs entry

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      memory: 200Mi
+tag_specification:
+  name: "4.8"
+  namespace: ocp
+tests:
+- as: e2e-metal-ipi
+  steps:
+    cluster_profile: packet
+    workflow: baremetalds-e2e
+- as: e2e-metal-ipi-virtualmedia
```
Comment (elfosardo) [inline ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main.yaml:39]: does this apply to virtualmedia scenarios?

**Review by elfosardo — CHANGES_REQUESTED** [review]
some changes and you need https://github.com/openshift/ironic-agent-image/pull/5 first

File: ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main.yaml (hunk 3 above)
Comment (dtantsur) [reply ci-operator/config/openshift/ironic-agent-image/openshift-ironic-agent-image-main.yaml:39]: We will switch everything to the new container.

**Comment by dtantsur** [comment]
/retest

I've reverted a part of the Dockerfile.ocp change

**Comment by dtantsur** [comment]
The images job has passed. The metal-ipi failures are irrelevant since the new project is not used in them yet.

**Review by elfosardo — APPROVED** [review]

**Pull request merged**